apis/r/R/utils-coercions.R

```diff
@@ -0,0 +1,40 @@
+#' Coercion methods for SOMA classes
+
+#' @importFrom arrow as_arrow_table
+#' @export
+as_arrow_table.SOMASparseNDArrayRead <- function(x){
+  x$tables()$concat()
+}
+
+#' Coerce \link[tiledbsoma]{SOMASparseNDArrayRead} to \link{data.frame} or \link[tibble]{tibble}
+#' @export
+as.data.frame.SOMASparseNDArrayRead <- function(x, ...){ 
+   as.data.frame(x$tables()$concat(), ...)
+}
+
+# Coerce \link[tiledbsoma]{SOMASparseNDArrayRead} to Matrix::\link[Matrix]{dgTMatrix}
+setAs(from = "SOMASparseNDArrayRead", 
+      to = "TsparseMatrix", 
+      def = function(from) from$sparse_matrix()$concat()
+      )
+
+# Coerce \link[tiledbsoma]{SOMASparseNDArrayRead} to Matrix::\link[Matrix]{dgCMatrix}
+setAs(from = "SOMASparseNDArrayRead", 
```

Comment:
We could also provide a delayed method for `SeuratObject::as.sparse()`; this is called in `SeuratObject::CreateSeuratObject()` and `SeuratObject::CreateAssayObject()` and would allow passing a sparse array/sparse array read directly to those functions

```R
#' @exportS3Method SeuratObject::as.sparse
#'
as.sparse.SOMASparseNDArrayRead <- function(x, ...) {
  as(x, "CsparseMatrix")
}
```